common/src/main/java/org/corfudb/common/metrics/micrometer/MeterRegistryProvider.java

```diff
@@ -79,15 +142,13 @@ private static void init(Supplier<Optional<MeterRegistry>> meterRegistrySupplier
      * @param type Type of a meter.
      */
     public static synchronized void deregisterServerMeter(String name, Tags tags, Meter.Type type) {
-        if (!meterRegistry.isPresent()) {
-            return;
-        }
         if (!id.isPresent()) {
-            throw new IllegalStateException("Id must be present to deregister meters.");
+            log.warn("Id must be present to deregister meters.");
+            return;
         }
         String server = id.get();
         Tags tagsToLookFor = tags.and(Tag.of("id", server));
         Meter.Id id = new Meter.Id(name, tagsToLookFor, null, null, type);
-        meterRegistry.ifPresent(registry -> registry.remove(id));
+        meterRegistry.remove(id);
```

Comment:
Local variable `id` has the same name with the global id?

Reply:
Fixed